packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -244,6 +265,27 @@ Future<String> _buildAotSnapshot(
 
   genSnapshotCmd.add(mainPath);
 
+  final File checksumFile = fs.file('$dependencies.checksum');
+  final List<File> checksumFiles = <File>[checksumFile, fs.file(dependencies)]
+      ..addAll(inputPaths.map(fs.file))
+      ..addAll(outputPaths.map(fs.file));
+  if (checksumFiles.every((File file) => file.existsSync())) {
```

Comment:
`exists()` checks are actually better sync anyway, since they won't hang and present less overhead than their async variants.